Core/include/Acts/Utilities/MathHelpers.hpp

```diff
@@ -27,4 +71,33 @@ constexpr auto fastHypot(T... args) {
   return std::sqrt(hypotSquare(args...));
 }
 
+/// @brief Calculates the sum of 1 + 2 + 3+ ... + N using the
+///        Gaussian sum formula
+/// @param N: Number until which the sum runs
+template <std::integral T>
+constexpr T sumAllNumUpTo(const T N) {
+  return N * (N + 1) / 2;
+}
+/// @brief Calculates the factorial of a number
+///        N!= N*(N-1)....*3*2*1
+/// @param upperN: Upper factor until which the factorial is calculated
+/// @param lowerN: Optional argument to remove the first factors from the calculation
+template <std::integral T>
+constexpr T factorial(const T upperN, const T lowerN = 1) {
```

Comment:
This could also have a shorter recursive version depending on how large the numbers are:

```cpp
template <typename T>
constexpr T factorial_constexpr(T n) {
    return (n <= 1) ? 1 : n * factorial_constexpr(n - 1);
}
```